### ISSUE TITLE
Turn off parse-numbers for a single entry in a \numlist command (Feature Request)

### ISSUE THREAD

**Issue by ihesham** [josephwright/siunitx#385]
I am using `\numlist` from the `siunitx` package to type a long list of numbers. I would like to turn off   `parse-numbers` for some individual entries so that a different number format would be allowed for these entries among the rest of the list entries that normally follow the default (or square brackets declared formats)
For example rendering an individual entry list of `1.` to being printed as `1.`  rather than `1.0`  by default. (A more detailed example of difficulty was posted in [TeX - LaTeX Stack Exchange](https://tex.stackexchange.com/q/548119/91556))

The feature that I request if possible is to allow individual brace protected entries in `\numlist` to skip being parsed, so that turning off parse-numbers for individual entries would become possible.

Thanks for your attention.

**Comment by josephwright** [comment]
This sounds like a recipe for inconsistent appearance ... can you explain why it's needed?

**Comment by josephwright** [comment]
In particular, I'd say the semantics of `\numlist` are that the entries are subject to the same formatting settings.

**Comment by ihesham** [comment]
> 
> 
> This sounds like a recipe for inconsistent appearance ... can you explain why it's needed?

Yes I can. I am preparing some notes for students about significant figures for which I am trying to explain that for example a list of all these numbers have four significant figures `100.0`, `1.000`, `1.000 \times 10^10`, `1000.`, etc. As you can see, the last number with 3 trailing zeros should be followed by a point to have a 4 significant figures. Typing it as it is in the `\numlist` entries renders it parsed to `1000.0` format which totally changes its significant figures into 5. Otherwise dropping the point would change it to `1000` which only has 1 significant figure.
